src/plugins/imagery/mixins/imageryData.js

```diff
@@ -44,8 +44,10 @@ export default {
         this.timeKey = this.timeSystem.key;
         this.timeFormatter = this.getFormatter(this.timeKey);
 
-        // kickoff
-        this.subscribe();
+        this.telemetryCollection = this.openmct.telemetry.requestCollection(this.domainObject, {});
+        this.telemetryCollection.on('add', this.dataAdded);
+        this.telemetryCollection.on('clear', this.dataCleared);
```

Comment:
Do we need to handle `remove` ?

Reply:
Is there a remove event for `telemetryCollection`?

Reply:
excellent catch again! i'm handling it now.